chain-abci/src/app/app_init.rs

```diff
@@ -235,17 +204,39 @@ impl ChainNodeApp {
                     &BitVec::from_elem(1, false).to_bytes(),
                 );
             }
-            inittx.put(COL_NODE_INFO, REWARDS_POOL_STATE_KEY, &rp.rlp_bytes());
-            inittx.put(
-                COL_NODE_INFO,
-                LAST_APP_HASH_KEY,
-                &genesis_app_hash.as_bytes(),
-            );
+
+            let last_state = match _req.time.as_ref() {
```

Comment:
`init_chain_handler()` and `new_with_storage()` have become too big for easy comprehension. Can these be broken down in multiple modular functions?

Reply:
broke them down a bit -- these parts will change, so we can refactor it more, as we go.
you can check if it's ok for now